src/sentry/static/sentry/app/views/organizationHealth/util/healthRequest.jsx

```diff
@@ -122,14 +123,41 @@ class HealthRequestWithParams extends React.Component {
   }
 
   componentDidUpdate(prevProps) {
-    if (isEqual(prevProps, this.props)) return;
+    const propNamesToCheck = [
+      'environments',
+      'includePrevious',
+      'includeTimeseries',
+      'includeTop',
+      'includeTransformedData',
+      'interval',
+      'limit',
+      'period',
+      'projects',
+      'showLoading',
+      'specifiers',
+      'tag',
+    ];
+
+    const prevPropsToCheck = pickBy(prevProps, (value, key) =>
+      propNamesToCheck.includes(key)
+    );
+
+    const propsToCheck = pickBy(this.props, (value, key) =>
+      propNamesToCheck.includes(key)
+    );
+
+    if (isEqual(prevPropsToCheck, propsToCheck)) return;
```

Comment:
Be a bit more aggressive when determining when to refetch data.

Reply:
Curious which of the other props is the one that's changing too much?

Reply:
I didn't look too much into it but we had things like API client instance... initially I was destructuring but it got a bit noisy, there's no good reason I picked `pickBy` instead of `omitBy` -- I think having an exclude list is better than having an include list (since we may need to add future request parameters)... thoughts?